File: src/Modules/Doctors/ErpClink.Modules.Doctors.Application/Doctors/Models/DoctorModels.cs
namespace ErpClink.Modules.Doctors.Application.Doctors.Models;

public sealed record RegisterDoctorRequest(
    string FirstName,
    string LastName,
    string? DisplayName,
    Guid? SpecialtyId,
    string? LicenseNumber,
    string? PhoneNumber,
    string? Email,
    string? UserId);

public sealed record UpdateDoctorRequest(
    string FirstName,
    string LastName,
    string? DisplayName,
    Guid? SpecialtyId,
    string? LicenseNumber,
    string? PhoneNumber,
    string? Email,
    string? UserId);

public sealed record SearchDoctorsRequest(
    string? Query,
    string? DoctorNumber,
    string? Name,
    string? Phone,
    string? Email,
    Guid? SpecialtyId,
    Guid? ClinicId,
    bool? IsActive,
    string? SortBy,
    bool SortDescending = false,
    int Page = 1,
    int PageSize = 20);

public sealed record AssignDoctorToClinicRequest(
    Guid ClinicId,
    DateOnly StartDate,
    DateOnly? EndDate);

public sealed record DoctorDto(
    Guid Id,
    Guid OrganizationId,
    Guid BranchId,
    string DoctorNumber,
    string? UserId,
    string FirstName,
    string LastName,
    string DisplayName,
    Guid? SpecialtyId,
    string? SpecialtyName,
    string? LicenseNumber,
    string? PhoneNumber,
    string? Email,
    bool IsActive,
    DateTime CreatedAtUtc,
    string? CreatedBy,
    DateTime? UpdatedAtUtc,
    string? UpdatedBy);

public sealed record DoctorListItemDto(
    Guid Id,
    string DoctorNumber,
    string DisplayName,
    Guid? SpecialtyId,
    string? SpecialtyName,
    string? PhoneNumber,
    string? Email,
    bool IsActive);

public sealed record DoctorClinicAssignmentDto(
    Guid Id,
    Guid DoctorId,
    Guid ClinicId,
    string ClinicCode,
    string ClinicName,
    DateOnly StartDate,
    DateOnly? EndDate,
    bool IsActive);

public sealed record PagedDoctorsResult(
    IReadOnlyList<DoctorListItemDto> Items,
    int Page,
    int PageSize,
    int TotalCount);

File: frontend/erpclink-web/src/app/features/doctors/models/doctor.models.ts
import { SimplePagedResult } from '../../../shared/models/api.models';

export interface RegisterDoctorRequest {
  firstName: string;
  lastName: string;
  displayName?: string | null;
  specialtyId?: string | null;
  licenseNumber?: string | null;
  phoneNumber?: string | null;
  email?: string | null;
  userId?: string | null;
}

export interface UpdateDoctorRequest {
  firstName: string;
  lastName: string;
  displayName?: string | null;
  specialtyId?: string | null;
  licenseNumber?: string | null;
  phoneNumber?: string | null;
  email?: string | null;
  userId?: string | null;
}

export interface SearchDoctorsParams {
  query?: string;
  doctorNumber?: string;
  name?: string;
  phone?: string;
  email?: string;
  specialtyId?: string;
  clinicId?: string;
  isActive?: boolean;
  sortBy?: string;
  sortDescending?: boolean;
  page?: number;
  pageSize?: number;
}

export interface AssignDoctorToClinicRequest {
  clinicId: string;
  startDate: string;
  endDate?: string | null;
}

export interface DoctorDto {
  id: string;
  organizationId: string;
  branchId: string;
  doctorNumber: string;
  userId: string | null;
  firstName: string;
  lastName: string;
  displayName: string;
  specialtyId: string | null;
  specialtyName: string | null;
  licenseNumber: string | null;
  phoneNumber: string | null;
  email: string | null;
  isActive: boolean;
  createdAtUtc: string;
  createdBy: string | null;
  updatedAtUtc: string | null;
  updatedBy: string | null;
}

export interface DoctorListItemDto {
  id: string;
  doctorNumber: string;
  displayName: string;
  specialtyId: string | null;
  specialtyName: string | null;
  phoneNumber: string | null;
  email: string | null;
  isActive: boolean;
}

export interface DoctorClinicAssignmentDto {
  id: string;
  doctorId: string;
  clinicId: string;
  clinicCode: string;
  clinicName: string;
  startDate: string;
  endDate: string | null;
  isActive: boolean;
}

export type PagedDoctorsResult = SimplePagedResult<DoctorListItemDto>;
